sdk/storage/azure-storage-blob/tests/test_blob_tags.py

```diff
@@ -98,33 +103,40 @@ def test_set_blob_tags(self, storage_account_name, storage_account_key):
         resp = blob_client.set_blob_tags(blob_tags)
 
         # Assert
-        self.assertIsNotNone(resp)
+        assert resp is not None
 
     @BlobPreparer()
-    def test_set_blob_tags_with_lease(self, storage_account_name, storage_account_key):
+    @recorded_by_proxy
+    def test_set_blob_tags_with_lease(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
         self._setup(storage_account_name, storage_account_key)
         blob_client, _ = self._create_block_blob()
-        lease = blob_client.acquire_lease()
+        lease = blob_client.acquire_lease(lease_id='00000000-1111-2222-3333-444444444444')
 
         # Act
         blob_tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
-        with self.assertRaises(HttpResponseError):
+        with pytest.raises(HttpResponseError):
             blob_client.set_blob_tags(blob_tags)
         blob_client.set_blob_tags(blob_tags, lease=lease)
 
         blob_client.get_blob_tags()
-        with self.assertRaises(HttpResponseError):
+        with pytest.raises(HttpResponseError):
             blob_client.get_blob_tags(lease="'d92e6954-3274-4715-811c-727ca7145303'")
         resp = blob_client.get_blob_tags(lease=lease)
 
-        self.assertIsNotNone(resp)
-        self.assertEqual(len(resp), 3)
+        assert resp is not None
+        assert len(resp) == 3
 
         blob_client.delete_blob(lease=lease)
 
     @pytest.mark.live_test_only
```

Comment:
Any idea why this is live only? I don't think it needs to be so can you try recording it?

Same with async version.

Reply:
`E           azure.core.exceptions.ServiceRequestError: <urllib3.connection.HTTPSConnection object at 0x0000023F39C954E0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed`
`ServiceRequestError` 😨

Reply:
Good catch! I originally tried but was failing in recorded (as always, it was because I forgot to put `recorded_by_proxy` so no recording was being generated 🤦